### DISCUSSION TITLE
[XFEM][Moving interface]

### DISCUSSION THREAD

**Discussion by coskrrb2002** [idaholab/moose#17162] [Q&A Modules: General] [answered]
Dear Users,

I have studied the moving interface in the XFEM app.

But I got some problems with domain.

I want to simulate an expansion of domain (block) and assign material properties in the expanded domain.

But it seems like there is no way to apply material properties to the expanded domain.

More precisely, how can I apply a moving interface between two different domains?

One domain will expand, and the other will be shrunk...

Until now, I could see examples for a single domain.

Best regards,
Nakkyu

**Comment by coskrrb2002** [comment]
I have read the following post.

https://github.com/idaholab/moose/pull/16008

Is it possible to use "ActivateElementUserObject" to change elements domain id from one to another?

If possible, I think I can use it to solve my problem.


Best regards,
Nakkyu

**Comment by aeslaughter** [comment, marked as answer]
@jiangwen84 Can you help out here?

**Reply by hugary1995** [reply]
@aeslaughter Do you want to review the PR #17101 which (together with another follow-up PR) does exactly this? @coskrrb2002 hopefully we can add this capability soon...

**Reply by coskrrb2002** [reply]
Thank you.
This is what I want to use for my modeling.
I guess I have to wait.

**Reply by jiangwen84** [reply]
@coskrrb2002  Could you let us what physical problem you are trying to solve?

**Reply by coskrrb2002** [reply]
I am trying to solve corrosion of metals in solution with oxide film formation.
So, that's why I need a domain growth and shrink~.
And I found the way to do this in moose just today.
So, I am applying it my model and waiting a result.
